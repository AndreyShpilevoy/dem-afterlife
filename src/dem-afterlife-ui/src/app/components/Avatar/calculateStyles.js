

const commonImagePart = {
    alignSelf: 'center',
    flex: '0 0 auto',
    margin: 'auto',
    maxHeight: '100%',
    maxWidth: '100%',
    objectFit: 'scale-down'
};

const calculateStyles = ({avatar}) => ({
    container: {
        alignItems: 'center',
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'center',
        width: avatar.size
    },
    avatar: {
        display: 'flex',
        flexDirection: 'row',
        height: avatar.size,
        justifyContent: 'center',
        marginLeft: 'auto',
        marginRight: 'auto',
        width: '100%',
        '&>img': {...commonImagePart}
    },
    avatarDefault: {
        backgroundImage: `url(${avatar.defaultImage})`,
        backgroundRepeat: 'no-repeat',
        backgroundSize: 'contain',
        minHeight: '100%',
        minWidth: '100%',
        '&>img': {...commonImagePart}
    }
});

export default calculateStyles;