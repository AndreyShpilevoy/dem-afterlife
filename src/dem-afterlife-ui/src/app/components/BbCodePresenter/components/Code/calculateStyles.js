const calculateStyles = themeName => ({
    code: {
        backgroundColor: '#59534D',
        border: '0.0625rem solid #9C9085',
        padding: 0.4
    },
    codeHeader: {
        color: '#D4D4D4',
        display: 'flex'
    },
    codeHeaderLeft: {
        marginRight: 'auto'
    },
    codeHeaderRight: {
        marginLeft: 'auto'
    },
    codeContent: {
        color: '#FFA100',
        fontFamily: 'monospace',
        whiteSpace: 'pre-wrap'
    },
    codeSelectButton: {
        color: '#CBC065',
        cursor: 'pointer',
        '&:hover': {
            color: '#FFF495'
        }
    },
    options: {meta: 'Code', themeName}
});

export default calculateStyles;