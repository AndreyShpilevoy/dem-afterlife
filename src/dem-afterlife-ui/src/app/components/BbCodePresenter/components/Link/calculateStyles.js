const calculateStyles = ({themeName}) => ({
    link: {
        wordBreak: 'break-all'
    },
    options: {meta: 'Link', themeName}
});

export default calculateStyles;