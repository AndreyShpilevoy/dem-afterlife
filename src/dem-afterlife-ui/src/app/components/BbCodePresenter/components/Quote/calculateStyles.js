const calculateStyles = ({themeName}) => ({
    quoteAuthor: {
        fontWeight: 'bold'
    },
    quote: {
        backgroundColor: '#9A8D7F',
        border: '0.0625rem solid #D4C5B6',
        margin: '0.3125rem 1.25rem',
        padding: 0.4
    },
    options: {meta: 'Quote', themeName}
});

export default calculateStyles;