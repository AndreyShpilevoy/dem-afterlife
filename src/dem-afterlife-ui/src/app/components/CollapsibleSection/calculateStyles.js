import {createMediaQueryMinMax} from 'utils';

const getCommonPart = ({general, header, body}) => ({
    '@global': {
        '@keyframes collapsibleSection-SlideDown': {
            from: {
                maxHeight: '0vh'
            },
            to: {
                maxHeight: '100vh'
            }
        },
        '@keyframes collapsibleSection-SlideUp': {
            from: {
                maxHeight: '100vh'
            },
            to: {
                maxHeight: '0vh'
            }
        }
    },
    general: {
        marginTop: general.marginTop
    },
    header: {
        backgroundColor: header.backgroundColor,
        color: header.color,
        height: header.height
    },
    titleClass: {
        paddingLeft: general.padding,
        display: 'block',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
        whiteSpace: 'nowrap'
    },
    headerHolder: {
        display: 'flex',
        alignItems: 'center'
    },
    headerText: {
        width: 'initial',
        fontSize: header.fontSize,
        overflow: 'hidden',
        textOverflow: 'ellipsis',
        whiteSpace: 'nowrap',
        height: '100%'
    },
    body: {
        backgroundColor: body.backgroundColor,
        color: body.color,
        paddingTop: general.padding / 2,
        paddingBottom: general.padding / 2,
        paddingLeft: general.padding,
        paddingRight: general.padding
    },
    termedSection: {
        width: '100%'
    },
    headerColumn: {
        display: 'flex',
        justifyContent: 'center'
    }
});

const getSizeSpecific = (grid, collapsibleSection) =>
    grid.containers.reduce((previous, {gridSize, mediaMinString, mediaMaxString}) => gridSize !== 'lg' && gridSize !== 'xl' ?
        {
            ...previous,
            ...createMediaQueryMinMax(
                mediaMinString,
                mediaMaxString,
                {
                    headerCursor: {
                        cursor: 'pointer'
                    },
                    headerArrowHolder: {
                        marginLeft: 'auto',
                        marginRight: 1
                    },
                    headerArrow: {
                        display: 'block',
                        width: collapsibleSection.header.iconSize,
                        height: collapsibleSection.header.iconSize,
                        transform: 'rotate(-90deg)',
                        transformOrigin: 'center center',
                        transition: 'transform 0.5s ease-in',
                        '&.closed': {
                            transform: 'rotate(0deg)'
                        },
                        '& > .SVGInline-svg': {
                            width: collapsibleSection.header.iconSize,
                            height: collapsibleSection.header.iconSize,
                            fill: collapsibleSection.header.color
                        }
                    },
                    bodyHolder: {
                        transformOrigin: 'top',
                        transition: collapsibleSection.body.transition,
                        animation: 'collapsibleSection-SlideDown 500ms ease-in',
                        '&.closed': {
                            animation: 'collapsibleSection-SlideUp 500ms ease-in',
                            transform: 'scaleY(0)',
                            maxHeight: 0
                        }
                    }
                }
            )
        } :
        previous,
    {});

const calculateStyles = ({themeName, grid, collapsibleSection}) => ({
    ...getCommonPart(collapsibleSection),
    ...getSizeSpecific(grid, collapsibleSection),
    ...{options: {meta: 'CollapsibleSection', themeName} }
});

export default calculateStyles;