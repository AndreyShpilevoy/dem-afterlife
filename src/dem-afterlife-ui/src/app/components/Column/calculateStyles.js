import {createMediaQueryMin} from 'utils';

const calculateColumnStyles = gridSize =>
    [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12].reduce((previous, current) => {
        const columnSize = `${8.333333333333334 * current}%`;
        return {
            ...previous,
            ...{
                [`col-${gridSize}-${current}`]: {
                    flexBasis: columnSize,
                    maxWidth: columnSize
                },
                [`col-${gridSize}Offset-${current}`]: {
                    marginLeft: columnSize
                }
            }
        };
    }, {});

const calculateStyles = ({grid}) =>
    grid.containers.reduce((previous, {gridSize, mediaMinString}) => (
        {...previous, ...createMediaQueryMin(mediaMinString, calculateColumnStyles(gridSize))}
    ), {});

export default calculateStyles;