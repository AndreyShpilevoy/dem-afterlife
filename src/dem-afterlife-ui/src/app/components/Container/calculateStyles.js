import {createMediaQueryMin} from 'utils';

const calculateStyles = ({themeName, grid}) =>
    grid.containers.reduce((previous, {width, mediaMinString}) => ({
        ...previous,
        ...createMediaQueryMin(mediaMinString, {
            container: {
                width,
                marginLeft: 'auto',
                marginRight: 'auto'
            }
        })
    }), {options: {meta: 'Container', themeName} });

export default calculateStyles;