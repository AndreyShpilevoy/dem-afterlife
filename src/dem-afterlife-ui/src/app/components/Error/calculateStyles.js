const calculateStyles = () => ({
    container: {
        width: 6.25,
        textAlign: 'center',
        borderColor: '#fb7c20',
        borderStyle: 'solid',
        borderWidth: 0.125
    },
    image: {
        width: 3,
        height: 3,
        '& > .SVGInline-svg': {
            width: 3,
            height: 3,
            fill: '#fb7c20'
        }
    }
});

export default calculateStyles;