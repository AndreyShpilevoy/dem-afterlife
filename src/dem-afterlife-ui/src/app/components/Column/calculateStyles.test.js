/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0, react/jsx-filename-extension:0 */

import calculateStyles from './calculateStyles';

describe('Column calculateStyles', () => {
    const defaultThemeObject = {
        themeName: 'default',
        grid: {
            containers: [
                {
                    gridSize: 'xs',
                    mediaMinString: 'min-width: 0px',
                    mediaMaxString: 'max-width: 575px',
                    width: '100%'
                },
                {
                    gridSize: 'sm',
                    mediaMinString: 'min-width: 576px',
                    mediaMaxString: 'max-width: 767px',
                    width: '540px'
                }
            ]
        }
    };

    it('should create expected object', () => {
        expect(calculateStyles(defaultThemeObject)).toMatchSnapshot();
    });
});
