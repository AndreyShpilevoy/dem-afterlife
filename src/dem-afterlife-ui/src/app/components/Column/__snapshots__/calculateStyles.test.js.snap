// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column calculateStyles should create expected object 1`] = `
Object {
  "@media (min-width: 0px)": Object {
    "col-xs-0": Object {
      "flexBasis": "0%",
      "maxWidth": "0%",
    },
    "col-xs-1": Object {
      "flexBasis": "8.333333333333334%",
      "maxWidth": "8.333333333333334%",
    },
    "col-xs-10": Object {
      "flexBasis": "83.33333333333334%",
      "maxWidth": "83.33333333333334%",
    },
    "col-xs-11": Object {
      "flexBasis": "91.66666666666667%",
      "maxWidth": "91.66666666666667%",
    },
    "col-xs-12": Object {
      "flexBasis": "100%",
      "maxWidth": "100%",
    },
    "col-xs-2": Object {
      "flexBasis": "16.666666666666668%",
      "maxWidth": "16.666666666666668%",
    },
    "col-xs-3": Object {
      "flexBasis": "25%",
      "maxWidth": "25%",
    },
    "col-xs-4": Object {
      "flexBasis": "33.333333333333336%",
      "maxWidth": "33.333333333333336%",
    },
    "col-xs-5": Object {
      "flexBasis": "41.66666666666667%",
      "maxWidth": "41.66666666666667%",
    },
    "col-xs-6": Object {
      "flexBasis": "50%",
      "maxWidth": "50%",
    },
    "col-xs-7": Object {
      "flexBasis": "58.333333333333336%",
      "maxWidth": "58.333333333333336%",
    },
    "col-xs-8": Object {
      "flexBasis": "66.66666666666667%",
      "maxWidth": "66.66666666666667%",
    },
    "col-xs-9": Object {
      "flexBasis": "75%",
      "maxWidth": "75%",
    },
    "col-xsOffset-0": Object {
      "marginLeft": "0%",
    },
    "col-xsOffset-1": Object {
      "marginLeft": "8.333333333333334%",
    },
    "col-xsOffset-10": Object {
      "marginLeft": "83.33333333333334%",
    },
    "col-xsOffset-11": Object {
      "marginLeft": "91.66666666666667%",
    },
    "col-xsOffset-12": Object {
      "marginLeft": "100%",
    },
    "col-xsOffset-2": Object {
      "marginLeft": "16.666666666666668%",
    },
    "col-xsOffset-3": Object {
      "marginLeft": "25%",
    },
    "col-xsOffset-4": Object {
      "marginLeft": "33.333333333333336%",
    },
    "col-xsOffset-5": Object {
      "marginLeft": "41.66666666666667%",
    },
    "col-xsOffset-6": Object {
      "marginLeft": "50%",
    },
    "col-xsOffset-7": Object {
      "marginLeft": "58.333333333333336%",
    },
    "col-xsOffset-8": Object {
      "marginLeft": "66.66666666666667%",
    },
    "col-xsOffset-9": Object {
      "marginLeft": "75%",
    },
  },
  "@media (min-width: 576px)": Object {
    "col-sm-0": Object {
      "flexBasis": "0%",
      "maxWidth": "0%",
    },
    "col-sm-1": Object {
      "flexBasis": "8.333333333333334%",
      "maxWidth": "8.333333333333334%",
    },
    "col-sm-10": Object {
      "flexBasis": "83.33333333333334%",
      "maxWidth": "83.33333333333334%",
    },
    "col-sm-11": Object {
      "flexBasis": "91.66666666666667%",
      "maxWidth": "91.66666666666667%",
    },
    "col-sm-12": Object {
      "flexBasis": "100%",
      "maxWidth": "100%",
    },
    "col-sm-2": Object {
      "flexBasis": "16.666666666666668%",
      "maxWidth": "16.666666666666668%",
    },
    "col-sm-3": Object {
      "flexBasis": "25%",
      "maxWidth": "25%",
    },
    "col-sm-4": Object {
      "flexBasis": "33.333333333333336%",
      "maxWidth": "33.333333333333336%",
    },
    "col-sm-5": Object {
      "flexBasis": "41.66666666666667%",
      "maxWidth": "41.66666666666667%",
    },
    "col-sm-6": Object {
      "flexBasis": "50%",
      "maxWidth": "50%",
    },
    "col-sm-7": Object {
      "flexBasis": "58.333333333333336%",
      "maxWidth": "58.333333333333336%",
    },
    "col-sm-8": Object {
      "flexBasis": "66.66666666666667%",
      "maxWidth": "66.66666666666667%",
    },
    "col-sm-9": Object {
      "flexBasis": "75%",
      "maxWidth": "75%",
    },
    "col-smOffset-0": Object {
      "marginLeft": "0%",
    },
    "col-smOffset-1": Object {
      "marginLeft": "8.333333333333334%",
    },
    "col-smOffset-10": Object {
      "marginLeft": "83.33333333333334%",
    },
    "col-smOffset-11": Object {
      "marginLeft": "91.66666666666667%",
    },
    "col-smOffset-12": Object {
      "marginLeft": "100%",
    },
    "col-smOffset-2": Object {
      "marginLeft": "16.666666666666668%",
    },
    "col-smOffset-3": Object {
      "marginLeft": "25%",
    },
    "col-smOffset-4": Object {
      "marginLeft": "33.333333333333336%",
    },
    "col-smOffset-5": Object {
      "marginLeft": "41.66666666666667%",
    },
    "col-smOffset-6": Object {
      "marginLeft": "50%",
    },
    "col-smOffset-7": Object {
      "marginLeft": "58.333333333333336%",
    },
    "col-smOffset-8": Object {
      "marginLeft": "66.66666666666667%",
    },
    "col-smOffset-9": Object {
      "marginLeft": "75%",
    },
  },
}
`;
